extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SyntheticBeanBuildItem.java

```diff
@@ -65,8 +84,9 @@ public SyntheticBeanBuildItem done() {
             if (supplier != null && runtimeValue != null) {
                 throw new IllegalStateException("It is not possible to specify both - a supplier and a runtime value");
             }
-            if (supplier == null && runtimeValue == null) {
-                throw new IllegalStateException("Either a supplier or a runtime value must be set");
+            if (creatorConsumer == null && supplier == null && runtimeValue == null) {
+                throw new IllegalStateException(
+                        "It is not defined how a bean instance should be created, use ExtendedBeanConfigurator#creator(), ExtendedBeanConfigurator#supplier() or ExtendedBeanConfigurator#runtimeValue()");
```

Comment:
```suggestion
                        "No creation modality has been defined for SyntheticBean of type " + implClazz + ", use ExtendedBeanConfigurator#creator(), ExtendedBeanConfigurator#supplier() or ExtendedBeanConfigurator#runtimeValue()");
```

Reply:
With all respect, I find _"No creation modality has been defined"_ really cryptic. Also the class info is not that important because the stack will contain the exact location of the failure and the extension developer should immediately fix the problem.

Reply:
Does it hurt to add the class info? If not, I'd add it. 
As for `No creation modality`, I agree that it's not the best wording. Let's see if I can find a better one.

Reply:
> Does it hurt to add the class info? If not, I'd add it.

It does not hurt but it's not that relevant in the context of a synthetic bean, e.g. if you add a synthetic bean for `java.lang.String` it does not help much.

I like the _"Synthetic bean does not provide a creation method..."_ though!